docs/user-guide/directconnection.rst

```diff
@@ -0,0 +1,178 @@
+############################
+Connecting directly to Kafka
+############################
+
+In some cases, you need to connect directly to Kafka to produce and consume messages (vs. going through the :ref:`rest-proxy`), like when publishing app metrics events, or when developing a `FastStream`_ app.
+This requires generating Kafka client credentials for the Sasquatch Kafka cluster and providing them to your app.
+
+Kafka provides `many different authentication options <https://docs.confluent.io/platform/current/security/authentication/overview.html>`__.
+This guide describes the the most secure and straightforward option, assuming that your app is running in the same Kubernetes cluster as the Sasquatch Kafka cluster.
+
+.. _FastStream: https://faststream.airt.ai/latest/
+
+Generating Kafka credentials
+============================
+
+You can generate Kafka credentials by creating a couple of `Strimzi`_ resources:
+
+* A `KafkaUser`_ resource, in the ``sasquatch`` namespace, to configure a user in the Kafka cluster and provision a Kubernetes Secret with that user's credentials
+* A `KafkaAccess`_ resource, in your app's namespace, to make those credentials and other Kafka connection information available to your app
+
+.. _Strimzi: https://strimzi.io
+.. _KafkaUser: https://strimzi.io/docs/operators/latest/configuring.html#type-KafkaUser-reference
+.. _KafkaAccess: https://github.com/strimzi/kafka-access-operator
+
+Strimzi KafkaUser resource
+--------------------------
+
+Here's an example of a ``KafkaUser`` with some of the common ACL rules you may want (more details about ACLs `here <https://docs.confluent.io/platform/current/security/authorization/acls/overview.html>`__):
+
+.. code-block:: yaml
+
+   apiVersion: kafka.strimzi.io/v1beta2
+   kind: KafkaUser
+   metadata:
+     name: myapp
+     labels:
+       # The name of the Strimzi ``Kafka`` resource, probably "sasquatch"
+       strimzi.io/cluster: sasquatch
+   spec:
+     template:
+         secret:
+           metadata:
+             annotations:
+               # These annotations allow the ``KafkaAccess`` resource to
+               # replicate values from the generated user credentials
+               # ``Secret`` into your app's namespace
+               replicator.v1.mittwald.de/replication-allowed: "true"
+               replicator.v1.mittwald.de/replication-allowed-namespaces: "myapp"
```

Comment:
Actually we can drop this. It was for a pre-Kafka access operator approach where I used Kubernetes secrets replicator to copy the kafka user's secrets.